### PR TITLE
Add circleci and azure pipelines support

### PR DESCRIPTION
**Pull request by npwalker**

### CODE REVIEW

File: moduleroot/azure-pipelines/spec_job.yml.erb [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  spec_command: ''
+  ruby_version: ''
+  puppet_gem_version: ''
+
+jobs:
+- job: ${{ parameters.name }}
+  variables:
+    puppet_gem_version: ${{ parameters.puppet_gem_version }}
+  pool:
+    vmImage: 'ubuntu-16.04'
```
Comment (glennsarti) [inline moduleroot/azure-pipelines/spec_job.yml.erb]: This should be parameterised.

File: moduleroot/azure-pipelines/spec_job.yml.erb [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      versionSpec: ${{ parameters.ruby_version }}
+      addToPath: true # Optional
+  - script: |
+      bundle -v
+      rm -f Gemfile.lock
+      gem update --system $RUBYGEMS_VERSION --no-document
+    displayName: 'gem update'
+  - script: ruby -v | tee ruby_version.txt
+    displayName: 'save ruby_version file'
+  - task: 1ESLighthouseEng.PipelineArtifactCaching.RestoreAndSaveCacheV1.RestoreAndSaveCache@1
```
Comment (glennsarti) [inline moduleroot/azure-pipelines/spec_job.yml.erb]: Perhaps this whole task should be wrapped in an `if` ?

File: moduleroot/.circleci/config.yml.erb [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+      puppet_gem_version: << parameters.puppet_gem_version >>
+    steps:
+      - runspecs:
+          spec_command: << parameters.spec_command >>
+
+workflows:
+  version: 2
+  PDK:
+    jobs:
+      - run_specs_on_ruby_puppet_combo:
```
Comment (glennsarti) [inline moduleroot/.circleci/config.yml.erb]: These jobs should be parameterised.  Similar to appveyor and travis

File: moduleroot/azure-pipelines/spec_job.yml.erb (hunk 2 above)
Comment (npwalker) [reply moduleroot/azure-pipelines/spec_job.yml.erb]: this task only runs if the condition is met.  So it is basically wrapped in an if statement.

Comment (glennsarti) [reply moduleroot/azure-pipelines/spec_job.yml.erb]: Fair enough

File: moduleroot/.circleci/config.yml.erb [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    environment:
+      PUPPET_GEM_VERSION: << parameters.puppet_gem_version >>
+
+commands:
+  runspecs:
+    description: "A command for running rake spec"
+    parameters:
+      spec_command:
+        type: string
+        #default: "parallel_spec"
```
Comment (glennsarti) [inline moduleroot/.circleci/config.yml.erb]: nit: remove

**Comment by scotje** [comment]
We should update this so that the config_defaults for these new configs are `deleted` or `unmanaged` so they are opt-in.

We also should note in the PDK release notes that these new configs are community supported.

**Comment by scotje** [comment]
Rebased and added the config_defaults to make the new configs unmanaged by default.

**Comment by scotje** [comment]
Hmmm, the `unmanage` behavior on a new module isn't really what we want I guess, but we don't want it to delete existing configs on an update either... :/ I think we're going to have to punt this to the next release and try to figure out a solution for opt-in templates.

**Comment by scotje** [comment]
This is blocked by PDK-1520 and some further investigation.

**Comment by npwalker** [comment]
What's the concern with including these by default?  All the other CI tool files are included by default and including them provides discoverability.

**Comment by daianamezdrea** [comment]
Hi @npwalker! Given the previous conversation, this PR will be closed. Thank you!

**Comment by npwalker** [comment]
hi @daianamezdrea what's the issue with adding these unmanaged by default?

**Comment by scotje** [comment]
I believe the issue is the current template renderer doesn't apply the "unmanaged" config option when rendering a new module. So the file would be laid down on a new module and then ignored on subsequent `update` operations.

**Pull request closed without merging**